lib/ansible/modules/windows/win_chocolatey_facts.ps1

```diff
@@ -0,0 +1,162 @@
+#!powershell
+
+# Copyright: (c) 2018, Simon Baerlocher <s.baerlocher@sbaerlocher.ch>
+# Copyright: (c) 2018, ITIGO AG <opensource@itigo.ch>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.CommandUtil
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version 2.0
+
+# Create a new result object
+$result = @{
+    changed = $false
+    ansible_facts = @{
+        ansible_chocolatey =  @{
+            config = @{}
+            feature = @{}
+            sources = @{}
+            packages = @{}
+        }
+    }
+}
+
+try {
+    $choco_app = Get-Command -Name choco.exe -CommandType Application -ErrorAction SilentlyContinue
+} catch {
+    Fail-Json -obj $result -message "Failed to find Chocolatey installation, make sure choco.exe is in the PATH env value"
+}
+
+Function Get-ChocolateyFeature {
+
+    param($choco_app)
+
+    $res = Run-Command -command "`"$($choco_app.Path)`" feature list -r"
+    if ($res.rc -ne 0) {
+        Fail-Json -obj $result -message "Failed to list Chocolatey features: $($res.stderr)"
+    }
+    $feature_info = @{}
+    $res.stdout -split "`r`n" | Where-Object { $_ -ne "" } | ForEach-Object {
+        $feature_split = $_ -split "\|"
+        $feature_info."$($feature_split[0])" = $feature_split[1] -eq "Enabled"
+    }
+
+    $result.ansible_facts.ansible_chocolatey.feature =  $feature_info
+
+}
+
+Function Get-ChocolateyConfig {
+
+    param($choco_app)
+
+    $choco_config_path = "$(Split-Path -Path (Split-Path -Path $choco_app.Path))\config\chocolatey.config"
+    if (-not (Test-Path -Path $choco_config_path)) {
+        Fail-Json -obj $result -message "Expecting Chocolatey config file to exist at '$choco_config_path'"
+    }
+
+    try {
+        [xml]$choco_config = Get-Content -Path $choco_config_path
+    } catch {
+        Fail-Json -obj $result -message "Failed to parse Chocolatey config file at '$choco_config_path': $($_.Exception.Message)"
+    }
+
+    $config_info = @{}
+    foreach ($config in $choco_config.chocolatey.config.GetEnumerator()) {
+        $config_info."$($config.key)" = $config.value
+    }
+
+    $result.ansible_facts.ansible_chocolatey.config =  $config_info
+}
+
+Function Get-ChocolateyPackages {
+
+    param($choco_app)
+
+    $res = Run-Command -command "`"$($choco_app.Path)`" list --local-only -r"
+    if ($res.rc -ne 0) {
+        Fail-Json -obj $result -message "Failed to list Chocolatey Packages: $($res.stderr)"
+    }
+
+    $packages_info = @{ }
+    $res.stdout -split "`r`n" | Where-Object { $_ -ne "" } | ForEach-Object {
+        $packages_split = $_ -split "\|"
+        $packages_info."$($packages_split[0])" = $packages_split[1]
```

Comment:
It's worthwile parsing this return value and actually setting a dict for each package with the key/value instead of relying on the user to know what's what.

Reply:
I would implement it that way, is that better?
```
    $packages_info = [System.Collections.ArrayList]@()
    $res.stdout -split "`r`n" | Where-Object { $_ -ne "" } | ForEach-Object {
        $packages_split = $_ -split "\|"
        $package_info = @{
            package = $packages_split[0]
            version = $packages_split[1]
         }
        $packages_info += $package_info
    }
```